src/stores/dialogStore.ts

```diff
@@ -0,0 +1,37 @@
+import { action, computed, observable } from 'mobx';
+import Constants from '../constants';
+
+export class DialogStore {
+    @observable private _message: string|null;
+
+    constructor() {
+        this._message = null;
+    }
+
+    @computed
+    get message() {
+        return this._message;
+    }
+
+    @computed
+    get isDialogOpen(): boolean {
+        return this._message !== null;
+    }
+
+    @action
+    public setFadeMessage = (message: string) => {
+        this._message = message;
+        setTimeout(() => {
+            this.close();
+        },         Constants.FADE_DIALOG_TIMEOUT);
```

Comment:
we could probably fix this indent by moving the start of the arrow function () to a new row.